src/crystal/system/dir.cr

```diff
@@ -53,6 +53,8 @@ end
 
 {% if flag?(:unix) %}
   require "./unix/dir"
+{% elsif flag?(:wasm32) %}
+  require "./wasm/dir"
```

Comment:
I believe most of these flags should actually be `wasi`.
`wasm32` is just the architecture, but these bindings are targeting the `wasi` system API.
And in fact, that should probably count as a `unix` environment since it's posix-like. So, when there are differences, specific branches for `wasi` need to be placed above generic `unix`.

```suggestion
{% if flag?(:wasi) %}
  require "./wasi/dir"
{% elsif flag?(:unix) %}
  require "./unix/dir"
```

Reply:
I made this change. Now `unix` includes `wasi` and all system impl is guarded by the `wasi` flag. It makes more sense.